tests/components/linknlink/__init__.py

```diff
@@ -0,0 +1,133 @@
+"""Tests for the LinknLnk integration."""
+
+from dataclasses import dataclass
+from unittest.mock import MagicMock, patch
+
+from homeassistant.components.linknlink.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+LINKNLINK_DEVICES = {
+    "Living Room": (
+        "192.168.2.12",
+        "ec0b12a43ba1",
+        "eHub",
+        "LinknLink",
+        "EHUB",
+        0x520B,
+        44017,
+        10,
+    ),
+    "Bedroom": (
+        "192.168.2.15",
+        "ec0b12a43bb2",
+        "eTHS",
+        "LinknLink",
+        "ETHS",
+        0xAC7C,
+        20025,
+        5,
+    ),
+    "Office": (  # Not supported.
+        "192.168.2.64",
+        "ec0b12a43bc3",
+        "eMotion",
+        "LinknLink",
+        "EMOTION",
+        0xAC7B,
+        57,
+        5,
+    ),
+    "Kitchen": (  # Not supported.
+        "192.168.2.64",
+        "ec0b12a43bc3",
+        "eMotion2",
+        "LinknLink",
+        "EMOTION2",
+        0x7777,
+        57,
+        5,
+    ),
+}
+
+
+@dataclass
+class MockSetup:
+    """Representation of a mock setup."""
+
+    api: MagicMock
+    entry: MockConfigEntry
+    factory: MagicMock
+
+
+class LinknLinkDevice:
+    """Representation of a LinknLink device."""
+
+    def __init__(
+        self, name, host, mac, model, manufacturer, type_, devtype, fwversion, timeout
+    ):
```

Comment:
Linter will want this to have a return of -> None

It is also probably worth it to type all of these parameters for readability.

Reply:
Okay - I swear linter used to yell at you if you didn't have a return type, but I tested it locally and it didn't complain, so maybe that changed? Or maybe i'm going crazy? It would still probably be best to return type everything though.

Reply:
Okay so that's only if you have strict typing enabled i think. https://developers.home-assistant.io/docs/development_typing/#including-modules-for-strict-type-checking